core/libs/commonwealth/commonwealth/utils/DHCPDiscovery.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+import asyncio
+import re
+import sys
+from typing import List
+
+from loguru import logger
+
+
+class DHCPDiscoveryError(Exception):
+    """Base exception for DHCP discovery errors"""
+
+
+async def discover_dhcp_servers(iface: str, timeout: int = 15) -> List[str]:
+    """
+    Discover DHCP servers on the network using nmap's DHCP discovery script
+
+    Args:
+        interface: Network interface to use
+        timeout: Time to wait for responses in seconds
+
+    Returns:
+        List of DHCP server IP addresses found
+
+    Raises:
+        DHCPDiscoveryError: If discovery fails
+    """
+    try:
+        # Run nmap with DHCP discovery script
+        cmd = [
+            "sudo",  # Need root privileges
+            "nmap",
+            "--script",
+            "broadcast-dhcp-discover",
+            "-e",
+            iface,  # Specify interface
+            "--script-timeout",
+            f"{timeout}s",
+        ]
+
+        # Run nmap asynchronously
+        logger.info(f"Running nmap command: {' '.join(cmd)}")
+        process = await asyncio.create_subprocess_exec(
+            *cmd, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+        )
+
+        stdout, stderr = await process.communicate()
+        output = stdout.decode()
+        error = stderr.decode()
+
+        if process.returncode != 0:
+            print(f"nmap output: {output}")
+            print(f"nmap error: {error}")
+            raise DHCPDiscoveryError(f"nmap failed: {stderr.decode()}")
```

Comment:
I think we can extract this into a function and share this code with [`core/services/autopilot_manager/mavlink_proxy/AbstractRouter.py:AbstractRouter::start`](https://github.com/bluerobotics/BlueOS/blob/1d6851042c65aea0fc893a7741140358d36ee6a9/core/services/ardupilot_manager/mavlink_proxy/AbstractRouter.py#L96) and [`core/services/helper/main.py:ping`](https://github.com/bluerobotics/BlueOS/blob/1d6851042c65aea0fc893a7741140358d36ee6a9/core/services/helper/main.py#L634).

Reply:
I'm trying not to touch other services as we are almost stable...

Reply:
Right, then we should prepare a PR with that change to be merged after stable

Reply:
I've just moved this to [this issue](https://github.com/bluerobotics/BlueOS/issues/3194).